pyuvdata/uvbeam/uvbeam.py

```diff
@@ -2471,6 +2582,15 @@ def __add__(
             pol_t2o = np.nonzero(
                 np.in1d(this.polarization_array, other.polarization_array)
             )[0]
+
+            if len(pnew_inds) > 0:
+                # if this does not have cross pols but other does promote to complex
+                cross_pols = [-3, -4, -7, -8]
+                if (
+                    np.intersect1d(other.polarization_array, cross_pols).size > 0
+                    and np.intersect1d(initial_pol_array, cross_pols).size == 0
+                ):
+                    this.data_array = np.asarray(this.data_array, dtype=complex)
```

Comment:
In testing this on my system, the datatype promotion with `dtype=complex` always resolved to `complex128`, even when the input was `float32`. This would effectively be a quadrupling of the required memory size, which may not be insignificant for large arrays. If we don't think it'll be an issue, we can keep it as is, but we may want to consider matching the data type better if we're expecting users to have single-precision arrays.

Reply:
If it's np.float32, promote to np.complex64, otherwise do this.